lib/streamlit/elements/multiselect.py

```diff
@@ -120,40 +132,7 @@ def multiselect(
         kwargs: Optional[WidgetKwargs] = None,
         *,  # keyword-only arguments:
         disabled: bool = False,
-    ) -> List[T]:
-        ...
-
-    @overload
-    def multiselect(
-        self,
-        label: str,
-        options: OptionSequence,
-        default: Optional[Any] = None,
-        format_func: Callable[[Any], Any] = str,
-        key: Optional[Key] = None,
-        help: Optional[str] = None,
-        on_change: Optional[WidgetCallback] = None,
-        args: Optional[WidgetArgs] = None,
-        kwargs: Optional[WidgetKwargs] = None,
-        *,  # keyword-only arguments:
-        disabled: bool = False,
-    ) -> List[Any]:
-        ...
-
-    def multiselect(
-        self,
-        label: str,
-        options: OptionSequence,
-        default: Optional[Any] = None,
-        format_func: Callable[[Any], Any] = str,
-        key: Optional[Key] = None,
-        help: Optional[str] = None,
-        on_change: Optional[WidgetCallback] = None,
-        args: Optional[WidgetArgs] = None,
-        kwargs: Optional[WidgetKwargs] = None,
-        *,  # keyword-only arguments:
-        disabled: bool = False,
-    ) -> List[Any]:
```

Comment:
The benefit of having `OptionSequence` be generic in all its variants can best be seen for `multiselect`, which was already properly typed, using the old definition of `OptionSequence`. Before, we needed two overload definitions to precisely state the return type. And we would have had to add those to the other widgets making use of the `OptionSequence` type. With the type being fully generic, we are able to get away with a single signature.